pulsar/main.py

```diff
@@ -334,6 +334,16 @@ def populate_options(cls, arg_parser):
         PulsarConfigBuilder.populate_options(arg_parser)
         arg_parser.add_argument("--manager", default=DEFAULT_MANAGER)
 
+    def load(self):
+        app_conf = super().load()
+        if self.manager != DEFAULT_MANAGER and "manager" in app_conf and "name" not in app_conf["manager"]:
+            log.info(
+                "Default manager not in app conf is not named but `--manager` is supplied, applying supplied manager "
+                f"name '{self.manager}' to default manager"
+            )
+            app_conf["manager"]["name"] = self.manager
+        return app_conf
```

Comment:
In puslar.py in the Galaxy job runner we have this in queue_job:

```
            remote_pulsar_app_config = dest_params.get("pulsar_app_config", {}).copy()
            if "pulsar_app_config_path" in dest_params:
                pulsar_app_config_path = dest_params["pulsar_app_config_path"]
                with open(pulsar_app_config_path) as fh:
                    remote_pulsar_app_config.update(yaml.safe_load(fh))
```

I feel like that might be the better place to put this? Or a better place.

Reply:
It looks like this block wouldn't be needed if you have a destination parameter declaring which manager to use - ```manager: tacc_k8s```. We've got code that is dispatching on the manager name and adding arguments at least:

```
    def _pulsar_script_args(self, manager_name, base64_job, base64_app_conf, wait_arg=None):
        manager_args = []
        if manager_name != "_default_":
            manager_args.append("--manager")
            manager_args.append(manager_name)
        if wait_arg:
            manager_args.append(wait_arg)
        manager_args.extend(["--base64", base64_job, "--app_conf_base64", base64_app_conf])
        return manager_args
```

The other block here is definitely good and I'll include a version of it in my PR.